procedures/table_exists.sql

```diff
@@ -0,0 +1,147 @@
+-- Copyright (c) 2014, Oracle and/or its affiliates. All rights reserved.
+-- 
+-- This program is free software; you can redistribute it and/or modify
+-- it under the terms of the GNU General Public License as published by
+-- the Free Software Foundation; version 2 of the License.
+-- 
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU General Public License for more details.
+-- 
+-- You should have received a copy of the GNU General Public License
+-- along with this program; if not, write to the Free Software
+-- Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+
+DROP PROCEDURE IF EXISTS table_exists;
+
+DELIMITER $$
+
+CREATE DEFINER='root'@'localhost' PROCEDURE table_exists (
```

Comment:
You don't set a SQL SECURITY characteristic with this, which essentially leads to some security leakage around knowing if a table exists or not for a user that is not root and has some limited set of privileges, because it will execute within the DEFINER role by default.

For the server itself, we generally respond with a "shrug, don't know what you're talking about" for a user with no privileges on an object that does exist. I think we should generally follow the same approach with this procedure as well - i.e, it should use "SQL SECURITY INVOKER", and the underlying selects will act just like the server does currently today anyway.